### PR TITLE
Crash preventation

### LINKED ISSUE #4659: Crash prevention in FIRCLSNetworkClient
- Xcode version: Version 11.3 (11C29)
- Firebase SDK version: deepens on component, not using the full sdk
- Firebase Component: Crashlytics
- Component version: Installed via cocoaPods

Hi everyone,
I would like to make a few changes in FIRCLSNetworkClient to make a code a little bit more safety :)
So the changes over here: https://github.com/firebase/firebase-ios-sdk/commit/bbc59fa45bca5c7de21925dfb32b776147f1ae66

I've seen that ReportManager passes nonnull block inside the networkClient but at some day any another client can pass a nil over here.

So may I create a pull request for this changes?
Also, FirebaseCrashlytics-Unit Tests successfully passed :)

### PR DESCRIPTION
**Pull request by GUserx**
Fixes #4659

### CODE REVIEW

File: Crashlytics/Crashlytics/Models/FIRCLSFileManager.m [hunk 1]
```diff
@@ -304,27 +306,14 @@ - (BOOL)removeContentsOfPendingPath {
...
-  }
-
+  BOOL contentsOfProcessingPathRemoved = [self removeContentsOfProcessingPath];
+  BOOL contentsOfPendingPathRemoved = [self removeContentsOfPendingPath];
+  BOOL contentsOfDirectoryAtPreparedPathRemoved =
+      [self removeContentsOfDirectoryAtPath:self.preparedPath];
+  BOOL contentsOfDirectoryAtActivePathRemoved =
+      [self removeContentsOfDirectoryAtPath:self.activePath];
+  success = contentsOfProcessingPathRemoved && contentsOfPendingPathRemoved &&
+            contentsOfDirectoryAtPreparedPathRemoved && contentsOfDirectoryAtActivePathRemoved;
```
Comment (GUserx) [inline Crashlytics/Crashlytics/Models/FIRCLSFileManager.m]: Here might be this type of form of code if you wish:
success = [self removeContentsOfProcessingPath] & [self removeContentsOfPendingPath] & ... etc
Imo above case is more friendly for debug =)

**Comment by paulb777** [comment]
@Guserx Thanks for the suggestions.

We'd appreciate if you'd send separate PRs for different pods. In this case, a separate PR for the Installations versus Crashylytics would help with the review process.

For the Installations change, please investigate the subsequent CI failure in https://travis-ci.org/firebase/firebase-ios-sdk/jobs/635952938

    - NOTE  | [iOS] xcodebuild:  /Users/travis/build/firebase/firebase-ios-sdk/Example/Messaging/Tests/FIRMessagingContextManagerServiceTest.m:85: error: -[FIRMessagingContextManagerServiceTest testMessageWithFutureStartTime] : ((self.scheduledLocalNotifications.count) equal to (1)) failed: ("0") is not equal to ("1")
    - NOTE  | [iOS] xcodebuild:  /Users/travis/build/firebase/firebase-ios-sdk/Example/Messaging/Tests/FIRMessagingContextManagerServiceTest.m:170: error: -[FIRMessagingContextManagerServiceTest testTimedNotificationsUserInfo] : ((self.scheduledLocalNotifications.count) equal to (1)) failed: ("0") is not equal to ("1")
    - NOTE  | [iOS] xcodebuild:  /Users/travis/build/firebase/firebase-ios-sdk/Example/Messaging/Tests/FIRMessagingContextManagerServiceTest.m:175: error: -[FIRMessagingContextManagerServiceTest testTimedNotificationsUserInfo] : ((notification.userInfo[messageIdentifierKey]) equal to (messageIdentifier)) failed: ("(null)") is not equal to ("fcm-cm-test1")
    - NOTE  | [iOS] xcodebuild:  /Users/travis/build/firebase/firebase-ios-sdk/Example/Messaging/Tests/FIRMessagingContextManagerServiceTest.m:176: error: -[FIRMessagingContextManagerServiceTest testTimedNotificationsUserInfo] : ((notification.userInfo[customDataKey]) equal to (customData)) failed: ("(null)") is not equal to ("world")

**Comment by GUserx** [comment]
> @GUserx Thanks for the suggestions.
> 
> We'd appreciate if you'd send separate PRs for different pods. In this case, a separate PR for the Installations versus Crashylytics would help with the review process.
> 
> For the Installations change, please investigate the subsequent CI failure in https://travis-ci.org/firebase/firebase-ios-sdk/jobs/635952938
> 
> ```
> - NOTE  | [iOS] xcodebuild:  /Users/travis/build/firebase/firebase-ios-sdk/Example/Messaging/Tests/FIRMessagingContextManagerServiceTest.m:85: error: -[FIRMessagingContextManagerServiceTest testMessageWithFutureStartTime] : ((self.scheduledLocalNotifications.count) equal to (1)) failed: ("0") is not equal to ("1")
> - NOTE  | [iOS] xcodebuild:  /Users/travis/build/firebase/firebase-ios-sdk/Example/Messaging/Tests/FIRMessagingContextManagerServiceTest.m:170: error: -[FIRMessagingContextManagerServiceTest testTimedNotificationsUserInfo] : ((self.scheduledLocalNotifications.count) equal to (1)) failed: ("0") is not equal to ("1")
> - NOTE  | [iOS] xcodebuild:  /Users/travis/build/firebase/firebase-ios-sdk/Example/Messaging/Tests/FIRMessagingContextManagerServiceTest.m:175: error: -[FIRMessagingContextManagerServiceTest testTimedNotificationsUserInfo] : ((notification.userInfo[messageIdentifierKey]) equal to (messageIdentifier)) failed: ("(null)") is not equal to ("fcm-cm-test1")
> - NOTE  | [iOS] xcodebuild:  /Users/travis/build/firebase/firebase-ios-sdk/Example/Messaging/Tests/FIRMessagingContextManagerServiceTest.m:176: error: -[FIRMessagingContextManagerServiceTest testTimedNotificationsUserInfo] : ((notification.userInfo[customDataKey]) equal to (customData)) failed: ("(null)") is not equal to ("world")
> ```

@paulb777 Thanks for your comment.
I did revert to previous version of this commit.

**Comment by paulb777** [comment]
Thanks. Note that CI did run for Crashlytics because of the issue fixed by #4663

**Comment by samedson** [comment]
@GUserx thanks for the PR! We're going to run through some validation and review this PR later in the week after we've gone through with the first release

**Comment by GUserx** [comment]
> @GUserx thanks for the PR! We're going to run through some validation and review this PR later in the week after we've gone through with the first release

Got it. Thank you for info.

File: Crashlytics/Crashlytics/Models/FIRCLSFileManager.m (hunk 1 above)
Comment (jasonhu-g) [reply Crashlytics/Crashlytics/Models/FIRCLSFileManager.m]: We can declare success at the time we check for all the criteria:
BOOL success = contentsOfProcessingPathRemoved && contentsOfPendingPathRemoved &&
	            contentsOfDirectoryAtPreparedPathRemoved && contentsOfDirectoryAtActivePathRemoved;

Comment (GUserx) [reply Crashlytics/Crashlytics/Models/FIRCLSFileManager.m]: > We can declare success at the time we check for all the criteria:
> BOOL success = contentsOfProcessingPathRemoved && contentsOfPendingPathRemoved &&
> contentsOfDirectoryAtPreparedPathRemoved && contentsOfDirectoryAtActivePathRemoved;

You are definitely right :)
Thanks!

**Pull request merged**